src/main/com/yetanalytics/lrs/xapi/statements.cljc

```diff
@@ -416,9 +416,100 @@
                :agents-only? (s/? boolean?))
   :ret (s/coll-of ::xs/actor))
 
+;; TODO: Deprecate
 (defn statements-equal? [& ss]
   (apply = (map dissoc-lrs-attrs ss)))
 
+;; NOTE: Case sensitivity is still applied to equality checks since the xAPI
+;; spec is vague on what string values are case insensitive.
+(defn- dissoc-statement-properties*
+  [substmt? stmt]
+  (let [{{stmt-act-type "objectType" :or {stmt-act-type "Agent"}}
```

Comment:
You use `:or` for missing/default `objectType` fields, but this won't ensure that it is present in the results, just that the `stmt-act-type` and `stmt-obj-type` bindings have a value. Will this fail on comparing two statements where both have activity objects but one specifies `object.objectType` and the other does not?

Reply:
I checked and it does not. That said, maybe the comparison _should_ fail since it's not explicitly said in the spec it's _not_ an immutable property.

Reply:
I think this comes down to the "social" choice: As an xAPI implementer using our LRS, which is more useful? I'd argue that it would be to consider them equal, as it should really be about differences in consequential properties like IDs and types.

Reply:
And in this case we _know_ the type is actually equal, even if one source doesn't take the time to specify it. Remember that xAPI implementers have to take multiple LRPs and point them at 1 LRS, so we should really err on the side of exposing more meaningful conflicts.

Reply:
Ok see: e4369b45c44776242f5b664cd01a7d8f80014771